sourcegraph/conn_pool.go

```diff
@@ -41,7 +41,7 @@ func pooledGRPCDial(target string, opts ...grpc.DialOption) (*grpc.ClientConn, e
 	if conns == nil {
 		conns = map[string]*grpc.ClientConn{}
 	}
-	if conn := conns[target]; conn != nil {
+	if conn := conns[target]; conn != nil && conn.State() != grpc.Shutdown {
```

Comment:
This seems reasonable. The other possible `conn.State()` values are:

https://godoc.org/google.golang.org/grpc#ConnectivityState

`grpc.Shutdown` seems to be the only one we'd want to take this action on.